src/Button/index.js

```diff
@@ -39,47 +31,31 @@ const Button = ({
       className={cc([
         "nds-typography",
         "nds-button",
-        `nds-button--${type}`,
+        `nds-button--${kind}`,
         {
           resetButton: as === "button",
           "nds-button--disabled": disabled,
         },
-        className,
       ])}
       disabled={isButtonElement && disabled ? true : undefined}
       data-testid="nds-button"
     >
-      <div className="nds-button-content">{buttonLabel}</div>
+      <div className="nds-button-content">{label}</div>
     </AsElement>
   );
 };
 
 Button.propTypes = {
+  /** Renders the button label */
+  label: PropTypes.string.isRequired,
   /** The html element to render as the root node of `Button` */
   as: PropTypes.oneOf(["a", "button"]),
-  /** Renders the button label */
-  label: PropTypes.string,
   /** disables the button when set to `true` */
   disabled: PropTypes.bool,
-  /** type of button to render */
-  type: PropTypes.oneOf(["primary", "secondary", "menu", "plain"]),
+  /** style of button to render */
+  kind: PropTypes.oneOf(["primary", "secondary", "menu", "plain"]),
   /** Click callback, with event object passed as argument */
   onClick: PropTypes.func,
-  /**
-   * ️**⚠️ DEPRECATED**
-   *
-   * Support for passing custom `className` strings will be removed in
-   * a future release.
-   * Please use the `type` prop to determine the button style instead.
-   */
-  className: PropTypes.string,
-  /**
-   * **⚠️ DEPRECATED**
-   *
-   * Passing children to render the button label will be removed
-   * in a future release. Use the `label` prop instead.
-   */
-  children: PropTypes.node,
```

Comment:
are you sure you want to do this? I thought we wanted nds usage to be as analogous to normal DOM elements as possible, and `<button>do thing</button>` is [current recommended practice](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#accessibility_concerns)

Reply:
> nds usage to be as analogous to normal DOM elements as possible

This may have been a stated goal before my time, but my goal is to make NDS components as reusable and composable as possible. The props interface does not need to match the DOM output to accomplish that.

As for the `button` element recommendation, I totally agree. This PR changes the default DOM element rendered to `button` (we were rendering an anchor up until now)

Reply:
@graphicsforge On closer, inspection it looks like we need to support an `icon` prop before we can do this:

`NewThreadModal.jsx`
```jsx
<Button
          className="ui primary button new-thread-button"
          onClick={(e) => onClick(e)}
        >
          <Plus size={17} />
          New message
        </Button>
```

I will roll back this change for the v2 release and `children` will be supported until we can add icon prop support to `Button`.